### PR TITLE
Fix race condition where trusted bundle is rendered twice with differ…

### PR DESCRIPTION
**Pull request by rene-dekker**
The trusted bundle is rendered twice in the elasticsearch namespace. Since all the required certificates for esmetrics should already be rendered before the esgw is rendered, we can simply refrain from re-rendering it.

### CODE REVIEW

File: pkg/controller/logstorage/esgateway.go
Comment (Brian-McM) [inline pkg/controller/logstorage/esgateway.go:77]: I don't think the prometheus cert is for es gateway, so I think this is a little misleading adding it here. I understand though why you've done it as other solutions would take longer.

I think since we're going towards having a single tigera ca config map, we'd need to develop a pattern where components can add what they need to the CA but don't actually render the secret for it (in case some other component needs to add more to the CA). This would probably involve a refactor of how we render all the elasticsearch components since they're essentially separate controllers.

Comment (Brian-McM) [reply pkg/controller/logstorage/esgateway.go:77]: If we're leaving this as is (which I think we are given the time constraints) could you please add a note here saying that this certificate is not for es-metrics and why it needs to be rendered here. Maybe also add a note in es metrics in case somebody is looking in es-metrics for the prometheus cert addition.

**Review by Brian-McM — APPROVED** [review]

File: pkg/controller/logstorage/esgateway.go
Comment (rene-dekker) [inline pkg/controller/logstorage/esgateway.go:134]: Changed the order so the secrets and cm are rendered before the deployments etc. The namespace is already rendered by logstorage.go

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
